scripts/mainsite.js

```diff
@@ -0,0 +1,213 @@
+    existingIframe.src = iframeUrl;
+    //Hide iframe when loading (to avoid white flashing screen)
+    existingIframe.style.visibility = "hidden";
+    existingIframe.onload = () => loadIframe(existingIframe);
+    container.appendChild(existingIframe);
+  }
+
+  lastStreamer = streamerName;
+});
+
+observer.observe(document.body, {
+  attributes: false,
+  childList: true,
+  subtree: true,
+});
+
+//Code modifying the iframe
+function loadIframe(iframe) {
+  let window = iframe.contentWindow;
+  let document = window.document;
+
+  let observer = new MutationObserver(function (mutations, observer) {
+    let logo = document.querySelector('[data-a-target="player-twitch-logo-button"]');
+    let card = document.getElementsByClassName("tw-card")[0];
+    let panel = document.getElementsByClassName("stream-info-social-panel")[0];
+    let fullscreenButton = document.querySelector('[data-a-target="player-fullscreen-button"]');
+
+    //If all the elements are loaded, we continue.
+    if (!logo || !card || !panel || !fullscreenButton)
+      return;
+      
+    //Once everything is loaded, we show the iframe
+    existingIframe.style.visibility = "";
+
+    //We have to make a function to get the real theater button since the element changes everytime you go in fullscreen mode
+    let getRealTheaterButton = () => window.parent.document.querySelector('[data-a-target="player-theatre-mode-button"]');
+
+    let realFullscreenButton = window.parent.document.querySelector('[data-a-target="player-fullscreen-button"]');
+
+    //We copy the button from the real interface (so we have the title in the good language and the good look)
+    let theaterButton = getRealTheaterButton().parentElement.cloneNode(true).getElementsByTagName("button")[0];
+
+
+    //No need to listen for changes now
+    observer.disconnect();
+
+    //We don't care about these elements
+    logo.remove();
+    panel.remove();
+
+    //When on fullscreen, we show the stream and streamer informations. Otherwise, we don't show them.
+    card.style.display = "none";//We hide them
+    window.parent.addEventListener("fullscreenchange", _ => {
+      card.style.display = window.parent.document.fullscreenElement ? "" : "none";
+      //We change the look of the fullscreen button when the fullscreen state change
+      fullscreenButton.innerHTML = realFullscreenButton.innerHTML;
+    });
+
+    let config = document.querySelector('[data-a-target="player-settings-button"]');
+    //The interface is not made in the same way using firefox or chrome
+    //Firefox way
+    if(fullscreenButton.parentElement.children.length > 3)
+      config.parentElement.parentElement.insertBefore(
+        theaterButton.parentElement,
+        config.parentElement.nextSibling
+      );
+    else//Chrome way
+      fullscreenButton.parentElement.parentElement.insertBefore(
+        theaterButton.parentElement,
+        fullscreenButton.parentElement
+      );

```

Comment:
```suggestion
const isFirefox = fullscreenButton.parentElement.children.length > 3; // could also use browser detection here

if (isFirefox) {
...
} else {
...
} 
```

This way conveys through code the idea instead of through the comments. In essence, it's cleaner.

Reply:
Thank you for all your suggestions, I'll look into this later since I don't have the time now (or if someone can do so before, it could be cool).

For the message handling thing, we are storing the latest resolution choosen by the user in the localstorage to reuse it when an iframe is append (we set the default resolution to the last used one).

We are loading the iframe only when the url should change, so it is not so often.

There is already some optimizations in place to avoid consuming too much cpu but maybe could be better.

The regex can be changed to a list, you're true